packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts

```diff
@@ -0,0 +1,220 @@
+   */
+  readonly hostname: string;
+  /**
+   * The service discovery namespace name, this is used for service discovery
+   *
+   * @example domain.local
+   */
+  readonly namespaceName: string;
+  /**
+   * Array of VirtualNodeBackendProps
+   */
+  readonly backends?: VirtualNodeBackendProps[];
+  /**
+   * Listener properties, such as portMappings and optional healthChecks
+   */
+  readonly listeners?: ListenerProps;
+}
+
+/**
+ * VritualNode represents a newly createded AppMesh VirtualNode
+ *
+ * @see https://docs.aws.amazon.com/app-mesh/latest/userguide/virtual_nodes.html
+ */
+export class VirtualNode extends cdk.Construct {
+  /**
+   * The name of the AppMesh which the virtual node belongs to
+   *
+   * @type {string}
+   * @memberof VirtualNode
+   */
+  public readonly meshName: string;
+  /**
+   * The name of the VirtualNode
+   *
+   * @type {string}
+   * @memberof VirtualNode
+   */
+  public readonly virtualNodeName: string;
+  /**
+   * The Amazon Resource Name belonging to the VirtualNdoe
+   *
+   * @type {string}
+   * @memberof VirtualNode
+   */
+  public readonly virtualNodeArn: string;
+
+  private readonly backends: CfnVirtualNode.BackendProperty[] = [];
+  private readonly listeners: CfnVirtualNode.ListenerProperty[] = [];
+  private readonly namespaceName: string;
+
+  constructor(scope: cdk.Construct, id: string, props: VirtualNodeProps) {
+    super(scope, id);
+
+    this.meshName = props.meshName;
+    this.namespaceName = props.namespaceName;
+
+    if (props.backends) {
+      this.addBackends(props.backends);
+    }
+
+    if (props.listeners) {
+      this.addListeners(props.listeners);
+    }
+
+    this.node.apply(new cdk.Tag(NAME_TAG, this.node.path));
+    const name = props.nodeName ? props.nodeName : id;
+
+    const node = new CfnVirtualNode(this, 'VirtualNode', {
+      virtualNodeName: name,
+      meshName: this.meshName,
+      spec: {
+        backends: this.backends,
+        listeners: this.listeners,
+        serviceDiscovery: {
+          dns: {
+            hostname: `${props.hostname}.${this.namespaceName}`,
+          },
+        },
+        logging: {
+          accessLog: {
+            file: {
+              path: '/dev/stdoout',
+            },
+          },
+        },
+      },
+    });
+
+    this.virtualNodeName = node.virtualNodeName;
+    this.virtualNodeArn = node.virtualNodeArn;
+  }
+
+  /**
+   * Utility method to add a single backend for existing or new VritualNodes
+   *
+   * @param {VirtualNodeBackendProps} props
+   * @memberof VirtualNode
+   */
+  public addBackend(props: VirtualNodeBackendProps) {
+    this.backends.push({
+      virtualService: {
+        virtualServiceName: props.virtualServiceName,
+      },
+    });
+  }
+
+  /**
+   * Utility method to add backends for existing or new VritualNodes
+   *
+   * @param {VirtualNodeBackendProps[]} props
+   * @memberof VirtualNode
+   */
+  public addBackends(props: VirtualNodeBackendProps[]) {
```

Comment:
I'm not sure this adds substantial value over `addBackend` itself.

Reply:
Other than preventing multiple calls to addBackend, it was meant as another option. I'm also not married to it. Do you think it hurts to keep around? It may be useful depending on the developer or devops engineer preferences. I'm always up for removing unnecessary code though, so let me know.

Reply:
I left this as is, for now, can easily be removed if you really want to. I'm not sure it hurts anything leaving there. Let me know and it can be removed.

Thanks